.github/workflows/api-dotnetcore.yml

```diff
@@ -98,18 +101,61 @@ jobs:
           fail_ci_if_error: true
 
       - name: SonarScanner for .NET 5 with pull request decoration support
+        id: scan
         uses: highbyte/sonarscan-dotnet@2.0
-        if: ${{ github.event_name == 'push' }}
+        if: github.event.pull_request.head.repo.full_name == github.repository 
         env:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         with:
           dotnetBuildArguments: ${{env.working-directory}}
           dotnetTestArguments: ${{env.working-directory}} --logger trx --collect:"XPlat Code Coverage" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover
           # Optional extra command arguments the the SonarScanner 'begin' command
-          sonarBeginArguments: /d:sonar.cs.opencover.reportsPaths="**/TestResults/**/coverage.opencover.xml" -d:sonar.cs.vstest.reportsPaths="**/TestResults/*.trx"
+          sonarBeginArguments: /d:sonar.cs.opencover.reportsPaths="**/TestResults/**/coverage.opencover.xml" -d:sonar.cs.vstest.reportsPaths="**/TestResults/*.trx" -d:sonar.qualitygate.wait=true
           # The key of the SonarQube project
-          sonarProjectKey: 483bc59d-f9ef-428c-866e-581d9ea69eab
+          sonarProjectKey: ${{secrets.PROJECT_KEY_API}}
           # The name of the SonarQube project
           sonarProjectName: PIMS-API
           # The SonarQube server URL. For SonarCloud, skip this setting.
           sonarHostname: ${{secrets.SONAR_URL}}
+          
+      - name: Find Comment
+        if: failure() && steps.scan.outcome == 'failure' && github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
+        uses: peter-evans/find-comment@v1
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: QUALITY GATE STATUS FOR .NET 5
+          
+      - name: Check Quality Gate and Create Comment
+        if: failure() && steps.scan.outcome == 'failure' && github.event_name == 'pull_request' && steps.fc.outputs.comment-id == '' && github.event.pull_request.head.repo.full_name == github.repository
+        uses: peter-evans/create-or-update-comment@v1
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_URL }}
+          PROJECT_KEY: ${{secrets.PROJECT_KEY_API}}
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            QUALITY GATE STATUS FOR .NET 5: FAILED.
+            
+            [View and resolve details on][1]
+            [1]: ${{env.SONAR_HOST_URL}}dashboard?id=${{env.PROJECT_KEY}}
+          reactions: confused
```

Comment:
@sookeke - unfortunately creating/updating comments do not work on pull requests triggered from forks. See https://github.com/peter-evans/create-or-update-comment. This will fail future PRs. This is consistent with what I saw with my auto-version action. The easiest solution is to use an admin token or personal access token - [PAT](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)